System/ContentFile.h

```diff
@@ -120,19 +113,19 @@ namespace RTE {
 		/// Gets the data represented by this ContentFile object as an Allegro BITMAP, loading it into the static maps if it's not already loaded. Note that ownership of the BITMAP is NOT transferred!
 		/// </summary>
 		/// <param name="conversionMode">The Allegro color conversion mode to use when loading this bitmap.</param>
-		/// <param name="storeBitmap">Whether to store the BITMAP in the relevant static map after loading it or not.</param>
+		/// <param name="storeBitmap">Whether to store the BITMAP in the relevant static map after loading it or not. If this is false, ownership of the BITMAP IS transferred!</param>
 		/// <param name="dataPathToSpecificFrame">Path to a specific frame when loading an animation to avoid overwriting the original preset DataPath when loading each frame.</param>
 		/// <returns>Pointer to the BITMAP loaded from disk.</returns>
 		BITMAP * GetAsBitmap(int conversionMode = 0, bool storeBitmap = true, const std::string &dataPathToSpecificFrame = "");
 
 		/// <summary>
-		/// Gets the data represented by this ContentFile object as an array of Allegro BITMAPs, each representing a frame in the animation.
-		/// It loads the BITMAPs into the static maps if they're not already loaded. Ownership of the BITMAPs is NOT transferred, but the array itself IS!
+		/// Gets the data represented by this ContentFile object as a vector of Allegro BITMAPs, each representing a frame in the animation.
+		/// It loads the BITMAPs into the static maps if they're not already loaded. Ownership of the BITMAPs is NOT transferred!
 		/// </summary>
-		/// <param name="frameCount">The number of frames to attempt to load, more than 1 frame will mean 00# is appended to datapath to handle naming conventions.</param>
+		/// <param name="frameCount">The number of frames to attempt to load, more than 1 frame will mean 00# is appended to DataPath to handle naming conventions.</param>
 		/// <param name="conversionMode">The Allegro color conversion mode to use when loading this bitmap.</param>
-		/// <returns>Pointer to the beginning of the array of BITMAP pointers loaded from the disk, the length of which is specified with the FrameCount argument.</returns>
-		BITMAP ** GetAsAnimation(int frameCount = 1, int conversionMode = 0);
+		/// <returns>Vector of BITMAP pointers loaded from the disk.</returns>
+		std::vector<BITMAP *> GetAsAnimation(int frameCount = 1, int conversionMode = 0);
```

Comment:
If you wanna maximize gainz you should make this return void (or a bool if you want) and take in the std::vector<BITMAP *> to fill as an argument.
Right now this method fills its vector internally then returns a copy of it, which means it has to do extra work copying. Using an out param would avoid that.

Reply:
Hmmmm. Fair point.

Reply:
I assume this is only really used during creation, not like things are gonna be constantly getting new animations or whatever, so it's not super important gainz, but wanted to mention it. Change it if you want to, or don't if you don't want to, not the end of the world either way.